flutter-studio/src/io/flutter/actions/ConnectAndroidDebuggerAction.java

```diff
@@ -5,158 +5,25 @@
-import io.flutter.run.SdkFields;
-import io.flutter.run.SdkRunConfig;
-import io.flutter.run.SdkAttachConfig;
 import io.flutter.sdk.FlutterSdkUtil;
 import org.jetbrains.android.actions.AndroidConnectDebuggerAction;
-import org.jetbrains.annotations.Nullable;
-
-import javax.swing.*;
-import java.util.ArrayList;
-import java.util.List;
 
 public class ConnectAndroidDebuggerAction extends AndroidConnectDebuggerAction {
-  @SuppressWarnings("FieldCanBeLocal")
-  private static String RUN_DEBUG_LINK = "https://flutter.io/using-ide/#running-and-debugging";
-
-  @Override
-  public void actionPerformed(AnActionEvent e) {
-    // NOTE: When making changes here, consider making similar changes to RunFlutterAction.
-    if (!FlutterUtils.isAndroidStudio()) {
-      super.actionPerformed(e);
-      return;
-    }
-    FlutterInitializer.sendAnalyticsAction(this);
-
-    RunnerAndConfigurationSettings settings = RunFlutterAction.getRunConfigSettings(e);
-    if (settings == null) {
-      showSelectConfigDialog();
-      return;
-    }
-
-    RunConfiguration configuration = settings.getConfiguration();
-    if (!(configuration instanceof SdkRunConfig)) {
-      Project project = e.getProject();
-      if (project == null || project.isDefault()) {
-        super.actionPerformed(e);
-        return;
-      }
-      if (!FlutterSdkUtil.hasFlutterModules(project)) {
-        super.actionPerformed(e);
-        return;
-      }
-      showSelectConfigDialog();
-      return;
-    }
-
-    SdkAttachConfig sdkRunConfig = new SdkAttachConfig((SdkRunConfig)configuration);
-    SdkFields fields = sdkRunConfig.getFields();
-    String additionalArgs = fields.getAdditionalArgs();
-
-    String flavorArg = null;
-    if (fields.getBuildFlavor() != null) {
-      flavorArg = "--flavor=" + fields.getBuildFlavor();
-    }
-
-    List<String> args = new ArrayList<>();
-    if (additionalArgs != null) {
-      args.add(additionalArgs);
-    }
-    if (flavorArg != null) {
-      args.add(flavorArg);
-    }
-    if (!args.isEmpty()) {
-      fields.setAdditionalArgs(Joiner.on(" ").join(args));
-    }
-
-    Executor executor = RunFlutterAction.getExecutor(ToolWindowId.DEBUG);
-    if (executor == null) {
-      return;
-    }
-
-    ExecutionEnvironmentBuilder builder = ExecutionEnvironmentBuilder.create(executor, sdkRunConfig);
-
-    ExecutionEnvironment env = builder.activeTarget().dataContext(e.getDataContext()).build();
-    FlutterLaunchMode.addToEnvironment(env, FlutterLaunchMode.DEBUG);
-
-    ProgramRunnerUtil.executeConfiguration(env, false, true);
-  }
 
   @Override
   public void update(AnActionEvent e) {
-    if (!FlutterUtils.isAndroidStudio()) {
-      super.update(e);
-      return;
-    }
     Project project = e.getProject();
     if (project == null || project.isDefault()) {
       super.update(e);
       return;
     }
-    if (!FlutterSdkUtil.hasFlutterModules(project)) {
-      super.update(e);
+    if (FlutterSdkUtil.hasFlutterModules(project)) {
```

Comment:
I'd consider not changing the default behavior here - not extending `ConnectAndroidDebuggerAction` - and just making sure that our icon + text are distinct from this action.

Reply:
That alternative isn't really great, I think. It would leave the toolbar cluttered with an unused but active button. When clicked it presents this confusing pane:

<img width="272" alt="screen shot 2018-10-03 at 12 57 14 pm" src="https://user-images.githubusercontent.com/8518285/46436402-95692b80-c70d-11e8-9b69-29378c956240.png">

I believe that's a poor UX. It's inviting issue creation, don't you think?

Reply:
It's definitely a place users could get caught up. We'd be balancing that vs potential issues - in the current AS/IntelliJ release and in future ones - that subclassing the action and (slightly) modifying its behavior causes issues.